tests/foreman/maintain/test_packages.py

```diff
@@ -0,0 +1,293 @@
+    result = sat_maintain.install(InstallerCommand('lock-package-versions'))
+    assert result.status == 0
+    assert 'Success!' in result.stdout
+
+    result = sat_maintain.cli.Packages.status()
+    assert 'Packages are locked.' in result.stdout
+    assert 'Automatic locking of package versions is enabled in installer.' in result.stdout
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+
+    result = sat_maintain.cli.Packages.is_locked()
+    assert 'Packages are locked' in result.stdout
+    assert result.status == 0
+
+    # Test whether packages are unlocked or not
+    result = sat_maintain.install(InstallerCommand('no-lock-package-versions'))
+    assert result.status == 0
+    assert 'Success!' in result.stdout
+
+    result = sat_maintain.cli.Packages.status()
+    assert 'FAIL' not in result.stdout
+    assert 'Packages are not locked.' in result.stdout
+    assert 'Automatic locking of package versions is disabled in installer.' in result.stdout
+    assert result.status == 0
+
+    result = sat_maintain.cli.Packages.is_locked()
+    assert 'Packages are not locked' in result.stdout
+    assert result.status == 1
+
+    # Teardown: lock packages of satellite, for capsule its unlocked by default
+    if sat_maintain.__class__.__name__ == 'Satellite':
+        result = sat_maintain.install(InstallerCommand('lock-package-versions'))
+        assert result.status == 0
+        assert 'Success!' in result.stdout
+
+
+@pytest.mark.include_capsule
+def test_positive_fm_packages_install(request, sat_maintain):
+    """Verify whether packages install/update work as expected.
+
+    :id: 645a3d84-34cb-469c-8b79-105b889aac78
+
+    :parametrized: yes
+
+    :steps:
+        1. Run satellite-installer --lock-package-versions
+        2. Run satellite-maintain packages status
+        3. Run satellite-maintain packages is-locked
+        4. Try to install/update package using satellite-maintain packages install/update command.
+        5. Run satellite-installer --no-lock-package-versions
+        6. Run satellite-maintain packages status
+        7. Run satellite-maintain packages is-locked
+        8. Try to install package in unlocked state.
+        9. Teardown (Run satellite-installer --lock-package-versions)
+
+    :expectedresults: Packages get install/update when lock/unlocked.
+    """
+    # Test whether packages are locked or not
+    result = sat_maintain.install(InstallerCommand('lock-package-versions'))
+    assert result.status == 0
+    assert 'Success!' in result.stdout
+
+    result = sat_maintain.cli.Packages.status()
+    assert 'Packages are locked.' in result.stdout
+    assert 'Automatic locking of package versions is enabled in installer.' in result.stdout
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+
+    result = sat_maintain.cli.Packages.is_locked()
+    assert 'Packages are locked' in result.stdout
+    assert result.status == 0
```

Comment:
I think this portion is repeating, we should straight jump to the actual agenda of this test.

Reply:
Or you are checking the locking of packages double unnecessary .

Reply:
yes, its just to exercise both f-m packages subcommand if they return correct output

Reply:
I mean the above test case is already testing it we don't need to reconfirm it in this test!